### PR TITLE
Discard any changes before we delete the persistent state

### PR DESCRIPTION
**Pull request by laevandus**
### 🎯 Goal

Make sure that changed managed object contexts do not trigger adding new data in `removeAllData`

### 📝 Summary

In the state layer branch we have a case where channel was not properly deleted on logout. That branch also has additional managed object context which is aggressively merging changes from other contexts (this seems to be related).

1. Log in with A
2. Open channel with A and B
3. Logout and log in with B
4. Open channel with A and B → issue happens (assert telling multiple ChannelDTOs for the same cid)

### 🛠 Implementation

Reset contexts when we logout for making sure we discard any active changes and do not try to save them when later the code path calls `writableContext.save`.

### 🎨 Showcase

From: `feature/chat-state-layer`
325224767-97655b0b-25bb-4976-9270-dabc040fe7e2

### 🧪 Manual Testing Notes

Follow the steps above.

### ☑️ Contributor Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] This change follows zero ⚠️ policy (required)
- [x] This change should be manually QAed
- [x] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (docusaurus, tutorial, CMS)

### CODE REVIEW

File: CHANGELOG.md
Comment (martinmitrevski) [inline CHANGELOG.md:8]: Shouldn't this be in the Fixed section?

**Review by martinmitrevski — APPROVED** [review]
LGTM ✅  Please check before merging the failing e2e tests (cc @testableapple)

**Comment by laevandus** [comment]
I will hold this one until we have released 4.53.0.

**Pull request merged**
